native/src/main/java/io/ballerina/stdlib/grpc/nativeimpl/client/FunctionUtils.java

```diff
@@ -114,6 +114,7 @@ public static Object externInit(BObject clientEndpoint, BString urlString,
         String scheme = url.getProtocol();
         Map<String, Object> properties =
                 HttpConnectorUtil.getTransportProperties(connectionManager.getTransportConfig());
+        properties.put(HttpConstants.CLIENT_CONFIG_HASH_CODE, clientEndpointConfig.toString().hashCode());
```

Comment:
Could you check the the difference of getting the string value of clientEndpointConfig record at ballerina level and Java level? IIRC in HTTP we got the ballerina level string value and got it passed to client creation stage.
I cant exactly remember the reason, but when we get the java level string, it always provided some different hash code.

Reply:
Fixed in https://github.com/ballerina-platform/module-ballerina-grpc/pull/623/commits/155dad75904cc7ee4cfb671e4d9df1ffc3480b1a